src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -116,12 +103,12 @@ class DefaultRedisCacheWriter implements RedisCacheWriter {
 		this.lockTtl = lockTtl;
 		this.statistics = cacheStatisticsCollector;
 		this.batchStrategy = batchStrategy;
+		this.asyncCacheWriter = isAsyncCacheSupportEnabled() ? new AsynchronousCacheWriterDelegate()
+				: UnsupportedAsyncCacheWriter.INSTANCE;
+	}
 
-		if (REACTIVE_REDIS_CONNECTION_FACTORY_PRESENT && this.connectionFactory instanceof ReactiveRedisConnectionFactory) {
-			asyncCacheWriter = new AsynchronousCacheWriterDelegate();
-		} else {
-			asyncCacheWriter = UnsupportedAsyncCacheWriter.INSTANCE;
-		}
+	private boolean isAsyncCacheSupportEnabled() {
+		return this.connectionFactory instanceof ReactiveRedisConnectionFactory;
```

Comment:
We're testing assignability against an interface that defines potentially absent types such as `Mono` and `Publisher`. Loading interfaces can work under certain circumstances, however depending on the runtime verifier mechanics, loading `ReactiveRedisConnectionFactory` without Project Reactor can cause `NoClassDefFoundError`.

Therefore, we used a safe scheme that is being used across the majority of portfolio projects and there's no reason to deviate from that.

Reply:
Understood

Reply:
Just thinking... perhaps the class presence check should be on a Reactive type (e.g. `Mono`) then, given it does depend on the runtime verifier, particularly in strange cases where it can load the `ReactiveConnectionFactory` interface, which has no explicit Reactive types until you drill into `ReactiveConnection`. The real question is, the `ReactiveConnection` as a return type on `ReactiveConnectionFactory` enough?

Also, by this same argument, then the only way a `ReactiveConnectionFactory` can exist is if _Lettuce_ is on the classpath anyway, which has a hard dependency on Reactor. Of course, there is no guarantee that Lettuce will always pull in Reactor as a transitive dependency (hence [this](https://github.com/spring-projects/spring-boot/blob/v3.1.4/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisReactiveAutoConfiguration.java#L43)), but also the reason I argued for [this](https://vmware.slack.com/archives/C055WU98P/p1694723156015419) (internal link).

Reply:
I convinced it should be both `ReactiveConnectionFactory` and a Reactive type like `Mono` or `Flux`.